relay.go

```diff
@@ -293,19 +293,29 @@ func (r *Relay) handleHopStream(s inet.Stream, msg *pb.CircuitRelay) {
 	// error, not an EOF.
 	go func() {
 		count, err := io.Copy(s, bs)
-		if err != io.EOF && err != nil {
+		if err != nil {
```

Comment:
so this will treat `io.EOF` as an error and reset. 
Is this really what we want? 
Or is it that `Copy` doesn't error with `EOF` at all?

Reply:
Copy doesn't return EOFs (it considers that a success case):

> A successful Copy returns err == nil, not err == EOF. Because Copy is defined to read from src until EOF, it does not treat an EOF from Read as an error to be reported.

Reply:
Cool, that's definitely an improvement.